backend/storage-service/src/jvmMain/kotlin/file/services/SynchronizationService.kt

```diff
@@ -0,0 +1,205 @@
+package dk.sdu.cloud.file.services
+
+import dk.sdu.cloud.Actor
+import dk.sdu.cloud.PageV2
+import dk.sdu.cloud.calls.RPCException
+import dk.sdu.cloud.file.LocalSyncthingDevice
+import dk.sdu.cloud.file.api.*
+import dk.sdu.cloud.file.services.acl.AclService
+import dk.sdu.cloud.file.synchronization.services.SyncthingClient
+import dk.sdu.cloud.service.db.async.*
+import io.ktor.http.*
+import java.io.File
+import java.util.*
+
+object SynchronizedFoldersTable : SQLTable("synchronized_folders") {
+    val id = varchar("id", 36, notNull = true)
+    val device = varchar("device_id", 64, notNull = true)
+    val path = text("path", notNull = true)
+    val accessType = varchar("access_type", 20, notNull = true)
+    val user = text("user_id", notNull = true)
+}
+
+object UserDevicesTable : SQLTable("user_devices") {
+    val device = varchar("device_id", 64, notNull = true)
+    val user = text("user_id", notNull = true)
+}
+
+class SynchronizationService(
+    private val syncthing: SyncthingClient,
+    private val fsPath: String,
+    private val db: DBContext,
+    private val aclService: AclService
+) {
+    private suspend fun chooseFolderDevice(): LocalSyncthingDevice? {
```

Comment:
Accept an already established session for this function. Right now this will open and close a database transaction for every single device instead of using the one that is likely already established.

Reply:
I suspect this code could benefit greatly from being cached for a little while.